safe_cli/main.py

```diff
@@ -132,14 +133,26 @@ def build_safe_cli():
         help="Enable history. By default it's disabled due to security reasons",
         default=False,
     )
+    parser.add_argument(
+        "--get-safes-from-owner",
+        action="store_true",
+        help="Indicates that address is an owner (Safe Transaction Service is required for this feature)",
+        default=False,
+    )
 
     args = parser.parse_args()
-
-    return SafeCli(args.safe_address, args.node_url, args.history)
+    if args.get_safes_from_owner:
+        if (
+            safe_address := get_safe_from_owner(args.address, args.node_url)
+        ) is not None:
+            return SafeCli(safe_address, args.node_url, args.history)
+    else:
+        return SafeCli(args.address, args.node_url, args.history)
 
 
 def main(*args, **kwargs):
-    safe_cli = build_safe_cli()
+    if (safe_cli := build_safe_cli()) is None:
```

Comment:
I think raising an exception would be more clear that just returning an empty `build_safe_cli`, as in the future we could have more reasons for building errors, what do you think?